src/Kiota.Builder/Writers/CSharp/CodeIndexerWriter.cs

```diff
@@ -11,10 +11,21 @@ public override void WriteCodeElement(CodeIndexer codeElement, LanguageWriter wr
         if (codeElement.Parent is not CodeClass parentClass) throw new InvalidOperationException("The parent of a property should be a class");
         var returnType = conventions.GetTypeString(codeElement.ReturnType, codeElement);
         conventions.WriteShortDescription(codeElement.Documentation.Description, writer);
+
+        MarkObsoleteIfDeprecationInformationIsPresent(codeElement, writer);
+        
         writer.StartBlock($"public {returnType} this[{conventions.GetTypeString(codeElement.IndexType, codeElement)} position] {{ get {{");
         if (parentClass.GetPropertyOfKind(CodePropertyKind.PathParameters) is CodeProperty pathParametersProp)
             conventions.AddParametersAssignment(writer, pathParametersProp.Type, pathParametersProp.Name.ToFirstCharacterUpperCase(), string.Empty, (codeElement.IndexType, codeElement.SerializationName, "position"));
         conventions.AddRequestBuilderBody(parentClass, returnType, writer, conventions.TempDictionaryVarName, "return ");
         writer.CloseBlock("} }");
     }
+
+    private static void MarkObsoleteIfDeprecationInformationIsPresent(CodeIndexer codeElement, LanguageWriter writer)
+    {
+        if (codeElement is { DeprecationInformation.Reason: { } deprecationReason })
+            writer.WriteLine($"""[System.Obsolete("{deprecationReason}")]""");
```

Comment:
Actually here you can call this extension method if you really want to make sure it's not going to contain any invalid character https://github.com/microsoft/kiota/blob/0521f7d667763a719c3be53d13e0a7e15b8dfcb9/src/Kiota.Builder/Extensions/StringExtensions.cs#L230